src/main/java/org/apache/pulsar/io/jcloud/format/JsonFormat.java

```diff
@@ -142,6 +146,20 @@ private Map<String, Object> convertRecordToObject(GenericRecord record) throws I
             return JSON_MAPPER.get().readValue((String) record.getNativeObject(), TYPEREF);
         } else if (record.getSchemaType() == SchemaType.BYTES) {
             return JSON_MAPPER.get().readValue((byte[]) record.getNativeObject(), TYPEREF);
+        } else if (record.getSchemaType() == SchemaType.KEY_VALUE) {
+            Map<String, Object> jsonKeyValue = new LinkedHashMap<>(2);
```

Comment:
Really a nit but if you want to avoid resizing the map, you would need to set `initialCapacity` to`2*0,75+1` because of the Map's loadFactor